projects/ad738x-st/src/examples/iio/iio_example.c

```diff
@@ -0,0 +1,92 @@
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdio.h>
+
+#include "iio_example.h"
+#include "common_data.h"
+#include "iio.h"
+#include "ad738x.h"
+#include "iio_ad738x.h"
+#include "no_os_util.h"
+#include "no_os_gpio.h"
+#include "no_os_print_log.h"
+#include "iio_app.h"
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief IIO example main execution.
+ *
+ * @return ret - Result of the example execution. If working correctly, will
+ *               execute continuously the while(1) loop and will not return.
+ */
+int iio_example_main()
+{
+	struct ad738x_iio_dev *dev;
+	struct iio_app_init_param app_init_param = {0};
+	struct iio_app_desc *app;
+	struct iio_app_device iio_devices[] = {
+		{ .name = "ad738x" },
+	};
+	int ret;
+
+	ret = ad738x_iio_init(&dev, &ad738x_ip);
+	if (ret)
+		return ret;
+
+	iio_devices[0].dev = dev;
+	iio_devices[0].dev_descriptor = dev->iio_dev;
+	app_init_param.devices = iio_devices;
+	app_init_param.nb_devices = NO_OS_ARRAY_SIZE(iio_devices);
+	app_init_param.uart_init_params = ad738x_uart_ip;
+
+	ret = iio_app_init(&app, app_init_param);
+	if (ret)
+		return ret;
+
+	return iio_app_run(app);
```

Comment:
before returning, in case of an error iio_app_remove should be called, and also ad7616_iio_remove. you may take a look here and follow the example: https://github.com/analogdevicesinc/no-OS/blob/master/projects/max22200/src/examples/iio_example/iio_example.c